packages/sdk/src/dataProtector/sharing/rentProtectedData.ts

```diff
@@ -0,0 +1,49 @@
+import { WorkflowError } from '../../utils/errors.js';
+import {
+  addressOrEnsOrAnySchema,
+  positiveNumberSchema,
+} from '../../utils/validators.js';
+import {
+  RentProtectedDataParams,
+  RentProtectedDataResponse,
+} from '../types.js';
+import { getSharingContract } from './smartContract/getSharingContract.js';
+
+export const rentProtectedData = async ({
+  collectionTokenId,
+  protectedDataAddress,
+}: RentProtectedDataParams): Promise<RentProtectedDataResponse> => {
+  try {
+    const vCollectionId = positiveNumberSchema()
+      .required()
+      .label('collectionId')
+      .validateSync(collectionTokenId);
+    const vProtectedDataAddress = addressOrEnsOrAnySchema()
+      .required()
+      .label('protectedDataAddress')
+      .validateSync(protectedDataAddress);
+    const sharingContract = await getSharingContract();
+    const rentingParams = await sharingContract.protectedDataForRenting(
+      vCollectionId,
+      vProtectedDataAddress
+    );
```

Comment:
This call can probably throw. Can we throw a specific error in this case?

Maybe also add a quick test to show what happens in case of this smart-contract call fails?

Reply:
i added a try catch to have a clear workflow error